zcncore/transaction.go

```diff
@@ -1298,26 +1303,60 @@ func (t *Transaction) MinerSCSharderSettings(info *MinerSCMinerInfo) (err error)
 	return
 }
 
-func (t *Transaction) MinerSCDeleteMiner(info *MinerSCMinerInfo) (err error) {
-	err = t.createSmartContractTxn(MinerSmartContractAddress,
-		transaction.MINERSC_MINER_DELETE, info, 0)
+func (ta *Transaction) MinerSCKillMiner(id string) error {
+	pid := ProviderId{
+		ID: id,
+	}
+	err := ta.createSmartContractTxn(MinerSmartContractAddress,
+		transaction.MINERSC_KILL_MINER, pid, 0)
 	if err != nil {
 		Logger.Error(err)
-		return
+		return err
 	}
-	go func() { t.setNonceAndSubmit() }()
-	return
+	go func() { ta.setNonceAndSubmit() }()
+	return nil
 }
 
-func (t *Transaction) MinerSCDeleteSharder(info *MinerSCMinerInfo) (err error) {
-	err = t.createSmartContractTxn(MinerSmartContractAddress,
-		transaction.MINERSC_SHARDER_DELETE, info, 0)
+func (ta *Transaction) MinerSCKillSharder(id string) error {
+	pid := ProviderId{
+		ID: id,
+	}
+	err := ta.createSmartContractTxn(MinerSmartContractAddress,
+		transaction.MINERSC_KILL_SHARDER, pid, 0)
 	if err != nil {
 		Logger.Error(err)
-		return
+		return err
 	}
-	go func() { t.setNonceAndSubmit() }()
-	return
+	go func() { ta.setNonceAndSubmit() }()
+	return nil
+}
+
+func (ta *Transaction) MinerSCShutDownMiner(id string) error {
+	pid := ProviderId{
```

Comment:
Could you be more explicit? None of the [transactions](https://github.com/0chain/gosdk/blob/staging/zcncore/transaction.go) have fee fields. `zwalletcli` methods in general, don't have an explicitly set a fee. TransactionScheme has a [setTranscationFee](https://github.com/0chain/gosdk/blob/staging/zcncore/transaction.go#L124) method, but it does not seem to be used.

Anyway, if it's a case that we should be adding explicit fee parameters to all zwallet clis then we should do this in a separate PR. 

Do we wish to maintain the TransactionScheme interface, which would probably involve calling  TransactionScheme.SetTransactionFee from zwallet? Or add a fee field to each TransactionScheme method.

Reply:
okay, let's do it in a separate PR if adding fee would break the transaction scheme interface.  @lpoli is working on the txn fee for clients, so perhaps he will fix this later.